cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -296,6 +297,13 @@ public StreamConfig call() throws Exception {
     });
   }
 
+  private void assertNamespaceHomeExists(Id.Namespace namespaceId) throws IOException {
+    Location namespaceHomeLocation = Locations.getParent(getStreamsHomeLocation(namespaceId));
+    Preconditions.checkArgument(namespaceHomeLocation.exists(),
+                                String.format("Expected home directory %s for namespace %s to exist.",
```

Comment:
I would say just copy the message from ``AppLifecycleHttpHandler`` or ``DatasetTypeHandler``